Document that Crowd2 plugin distribution will be suspended

MarkEWaite opened the pull request:
## Document that plugin distribution will be suspended

The Crowd 2 integration plugin is using dependencies that are closed source and are not licensed with an OSI approved open source license.

The [Jenkins project governance document](https://www.jenkins.io/project/governance/#license) and the [plugin hosting page](https://www.jenkins.io/doc/developer/publishing/preparation/#license) state that we only deliver plugins with open source licenses.

The specific license of concern is:

* [Atlassian customer license](https://www.atlassian.com/legal/software-license-agreement)

The same license of concern is used for:

* [Atlassian 3.0 End User License Agreement](http://www.atlassian.com/end-user-agreement/)

From the `/plugin/crowd2/wrapper/thirdPartyLicenses` page of a Jenkins installation that includes the Crowd 2 plugin:

Dependencies using the Atlassian Customer Agreement:

* com.atlassian.crowd:crowd-integration-api:5.1.5
* com.atlassian.crowd:crowd-integration-client-rest:5.1.5
* com.atlassian.crowd:embedded-crowd-api:5.1.5
* com.atlassian.crowd:crowd-integration-client-common:5.1.5

Dependencies using the Atlassian 3.0 End User License Agreement

* com.atlassian.collectors:atlassian-collectors-util:1.1

### Related pull requests and issues

* https://github.com/jenkins-infra/update-center2/pull/755
* https://github.com/jenkins-infra/helpdesk/issues/3842
* https://github.com/jenkinsci/configuration-as-code-plugin/pull/2433

### Testing done

Confirmed that the page layout of the documentation is good.

Confirmed that Jenkins 2.426.1 reports those licenses as displayed here:

![plugin-crowd2-wrapper-thirdPartyLicenses-2023-12-08-10_19_12](https://github.com/jenkinsci/crowd2-plugin/assets/156685/8391d192-32c2-4305-8cd9-5be2a283d1a5)

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes

MarkEWaite committed:
Document that plugin distribution will be suspended

The Crowd 2 integration plugin is using dependencies that are closed
source and are not licensed with an OSI approved open source license.

The Jenkins project governance document and the plugin hosting page
state that we only deliver plugins with open source licenses.

The specific license of concern is:

* https://www.atlassian.com/legal/software-license-agreement
  Atlassian customer license

The same license of concern is used for:

* http://www.atlassian.com/end-user-agreement/
  Atlassian 3.0 End User License Agreement

From the `/plugin/crowd2/wrapper/thirdPartyLicenses` page of a Jenkins
installation that includes the Crowd 2 plugin:

Dependencies using the Atlassian Customer Agreement:

* com.atlassian.crowd:crowd-integration-api:5.1.5
* com.atlassian.crowd:crowd-integration-client-rest:5.1.5
* com.atlassian.crowd:embedded-crowd-api:5.1.5
* com.atlassian.crowd:crowd-integration-client-common:5.1.5

Dependencies using the Atlassian 3.0 End User License Agreement

* com.atlassian.collectors:atlassian-collectors-util:1.1

DuMaM reviewed (approved):
Hi Mark,

Could you please add this notice too:
https://www.atlassian.com/licensing/crowd#crowd-data-center

Thanks

MarkEWaite committed:
Link to Crowd Server end of life announcement

https://www.atlassian.com/licensing/crowd#crowd-data-center says:

Important changes to our Server and Data Center products

We’ve ended sales for new server licenses and will end support for
server on February 15, 2024 PT. We’re continuing investment in Data
Center with several key improvements.

https://www.atlassian.com/migration/assess/journey-to-cloud provides a
migration path.

MarkEWaite commented:
> Hi Mark,
> 
> Could you please add this notice too: https://www.atlassian.com/licensing/crowd#crowd-data-center
> 
> Thanks

Done in 45bbb08f42680f82707d2daf5a77e8fefc185d25.  Let me know if that's not sufficient.

DuMaM commented:
All good thanks
